.github/workflows/main.yml

```diff
@@ -0,0 +1,62 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  industrial_ci:
+    name: ${{ matrix.env.CI_NAME }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        env:
+          - {CI_NAME: Bionic-Build,
+              OS_NAME: ubuntu,
+              OS_CODE_NAME: bionic,
+              ROS_DISTRO: melodic,
+              ROS_REPO: main,
+              UPSTREAM_WORKSPACE: "github:ros-industrial-consortium/tesseract#master github:ros-industrial-consortium/trajopt_ros#master
+              github:swri-robotics/descartes_light#master github:Jmeyer1292/opw_kinematics#master github:ethz-adrl/ifopt#master",
+              ROSDEP_SKIP_KEYS: "bullet3 fcl",
+              DOCKER_IMAGE: "lharmstrong/tesseract:melodic",
+              PARALLEL_TESTS: false,
+              CCACHE_DIR: "/home/runner/work/trajopt_ros/trajopt_ros/Bionic-Build/.ccache",
+              UPSTREAM_CMAKE_ARGS: "-DCMAKE_BUILD_TYPE=Release",
+              TARGET_CMAKE_ARGS: "-DCMAKE_BUILD_TYPE=Debug"}
+          - {CI_NAME: Focal-Build,
+              OS_NAME: ubuntu,
+              OS_CODE_NAME: focal,
+              ROS_DISTRO: noetic,
+              ROS_REPO: main,
+              UPSTREAM_WORKSPACE: "github:brisa-robotics/tesseract#noetic github:ros-industrial-consortium/trajopt_ros#master
```

Comment:
To address the package.xml differences, we may be able to use something like. I've never used and/or in a package.xml but it [looks](https://www.ros.org/reps/rep-0149.html) like it should work
```
  <depend condition="$ROS_VERSION == 1 and $ROS_DISTRO != noetic">orocos_kdl</depend>
  <depend condition="$ROS_VERSION == 2 or  $ROS_DISTRO == noetic">liborocos-kdl-dev</depend>
```

and 
```
  <depend condition="$ROS_PYTHON_VERSION == 2">python</depend>
  <depend condition="$ROS_PYTHON_VERSION == 3">python3</depend>
```

@Levi-Armstrong may have a cleaner solution though.

Reply:
I didn't know about this, that's great. I will patch the branch as soon as Levi answers.

Reply:
I have created a [PR](https://github.com/ros-industrial-consortium/tesseract/pull/319) against Tesseract adding the conditionals but it appear to break industrial_ci for some reason. I have created an [issue](https://github.com/ros-industrial/industrial_ci/issues/546) on industrial_ci though not sure why this would have caused an issue.

Reply:
[Tesseract #319](https://github.com/ros-industrial-consortium/tesseract/pull/319) should have fixed this. You should be able to switch it back to upstream master.

Reply:
This seems to work. Thanks.